Improve presence API

Y-Sindo opened the pull request:
* Expose a dedicated model class `SignalRGroupConnection` instead of `GroupMember` from SignalR service protocol.

vicancy commented on src/Microsoft.Azure.SignalR.Common/Pagination/GroupMemberQueryResultPage.cs:
```diff
@@ -34,3 +32,58 @@ public override Response GetRawResponse()
     }
 }
+
+internal class GroupMemberQueryResultPageConverter : JsonConverter<GroupMemberQueryResultPage>
```
why not simply use `[JsonProperty("value")]`/`[JsonProperty("nextlink")]`

Y-Sindo replied:
If we use `JsonProperty("value")`, we need to load the JSON object into memory first, but in this way, we don't need to load the JSON object, just process it streamingly.

vicancy commented on src/Microsoft.Azure.SignalR.Management/HubContext/GroupManager.cs:
```diff
@@ -50,6 +49,6 @@ public abstract class GroupManager : IGroupManager
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>An asynchronous enumerable of group members.</returns>
-        internal virtual AsyncPageable<GroupMember> ListConnectionsInGroup(string groupName, int? top = null, CancellationToken cancellationToken = default) => throw new NotImplementedException();
+        internal virtual AsyncPageable<SignalRGroupConnection> ListInGroup(string groupName, int? top = null, CancellationToken cancellationToken = default) => throw new NotImplementedException();
```
I would prefer `GetGroupConnections`

Y-Sindo committed:
Expose dedicated `SignalRGroupConnection`  model for presence API

Expose a dedicated model class `SignalRGroupConnection` instead of `GroupMember` from SignalR service protocol.

Y-Sindo committed:
fix

Y-Sindo committed:
fix

vicancy commented on src/Microsoft.Azure.SignalR.Common/Pagination/GroupMemberQueryResultPage.cs:
```diff
@@ -34,3 +32,58 @@ public override Response GetRawResponse()
+            string propertyName = reader.GetString()!;
+            reader.Read();
+
+            switch (propertyName.ToLowerInvariant())
```
avoid allocation?
```cs
switch (propertyName)
{
    case string s when s.Equals("value", StringComparison.OrdinalIgnoreCase):

Y-Sindo committed:
fix
